flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImplTest.java

```diff
@@ -1301,13 +1305,14 @@ private SlotRequest createSlotRequest(JobID jobId, ResourceProfile resourceProfi
 	}
 
 	private SlotManagerImpl createSlotManager(ResourceManagerId resourceManagerId, ResourceActions resourceManagerActions) {
-		return createSlotManager(resourceManagerId, resourceManagerActions, 1);
+		return createSlotManager(resourceManagerId, resourceManagerActions, new Configuration());
 	}
 
-	private SlotManagerImpl createSlotManager(ResourceManagerId resourceManagerId, ResourceActions resourceManagerActions, int numSlotsPerWorker) {
+	private SlotManagerImpl createSlotManager(ResourceManagerId resourceManagerId, ResourceActions resourceManagerActions, Configuration configuration) {
```

Comment:
nit: I would rather introduce `SlotManagerBuilder#buildAndStart[WithDirectExec](resourceManagerId[, executor], resourceManagerActions)`.

Reply:
Sounds good.

Reply:
However, to avoid code duplication, it seems that we still need these two `createSlotManager`, right?

Reply:
You could have a method to return a common builder for this test suite `SlotManagerBuilder.newBuilder().setDefaultWorkerResourceSpec(WORKER_RESOURCE_SPEC)`.

I would then use explicitly other setters and `buildWithDirectExec`.
The reason we use builders is to improve readability to avoid methods with a lot of parameters which are not named in java. Setters basically make them named for the price of some duplication and verbosity.

I would also agree to keep the existing `createSlotManager` with 3 args (`NUM_TASK_SLOTS`) but adding a 4th arg for `MAX_SLOT_NUM` may be too much indeed.

imo, passing `Configuration` makes some implicit contract which is not obvious that it is used only for `NUM_TASK_SLOTS` and `MAX_SLOT_NUM`.